pkg/ccl/backupccl/backup_job.go

```diff
@@ -199,37 +206,59 @@ func backup(
 	evalCtx := execCtx.ExtendedEvalContext()
 	dsp := execCtx.DistSQLPlanner()
 
-	// We don't return the compatible nodes here since PartitionSpans will
-	// filter out incompatible nodes.
-	planCtx, _, err := dsp.SetupAllNodesPlanningWithOracle(
-		ctx, evalCtx, execCtx.ExecCfg(), physicalplan.DefaultReplicaChooser, execLocality,
-	)
-	if err != nil {
-		return roachpb.RowCount{}, 0, errors.Wrap(err, "failed to determine nodes on which to run")
-	}
-
 	job := resumer.job
-	backupSpecs, err := distBackupPlanSpecs(
-		ctx,
-		planCtx,
-		execCtx,
-		dsp,
-		int64(job.ID()),
-		spans,
-		introducedSpans,
-		pkIDs,
-		defaultURI,
-		urisByLocalityKV,
-		encryption,
-		&kmsEnv,
-		kvpb.MVCCFilter(backupManifest.MVCCFilter),
-		backupManifest.StartTime,
-		backupManifest.EndTime,
-	)
-	if err != nil {
+	var backupSpecs map[base.SQLInstanceID]*execinfrapb.BackupDataSpec
+
+	// Note that planCtx is going to close over txn in db.Txn closure below, so it
+	// will still contain a pointer to that txn even after it commits when we pass
+	// planCtx to dsp.Run below. However dsp.Run does not use the txn in planCtx,
+	// since it is passed its own txn (which will be nil), so the fact it is an
+	// already-committed txn is not a problem.
+	var planCtx *sql.PlanningCtx
+
+	// We need to make a txn to setup planning since the txn is what will carry
+	// the read-as-of timestamp to the replica choice function, which needs it to
+	// determine if follower reads are possible. This txn is not intended to be
+	// used for anything -- no reads, no writes -- other than carrying its fixed
+	// timestamp from here to the replica choice oracle.
+	if err := execCtx.ExecCfg().DB.Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
+		var planningTxn *kv.Txn
+		if backupUseFollowerReads.Get(execCtx.ExecCfg().SV()) {
+			if err := txn.SetFixedTimestamp(ctx, backupManifest.EndTime); err != nil {
+				return err
+			}
+			planningTxn = txn
+		}
+		// We don't return the compatible nodes here since PartitionSpans will
+		// filter out incompatible nodes.
+		planCtx, _, err = dsp.SetupAllNodesPlanningWithOracleAndFollowers(
+			ctx, evalCtx, execCtx.ExecCfg(), physicalplan.DefaultReplicaChooser, execLocality, planningTxn,
+		)
+		if err != nil {
+			return errors.Wrap(err, "failed to determine nodes on which to run")
+		}
+
+		backupSpecs, err = distBackupPlanSpecs(
```

Comment:
It should be fine since we aren't _doing anything_ in the txn, but I'll note for visibility that in the case of encrypted backups this will be doing an external network call inside the txn.